Update Sweden holidays: add legally-recognized non-public holidays as `DE_FACTO` category

vruss opened the pull request:
## Proposed change

This PR introduces a new `DE_FACTO` holiday category to address issue #3118 where `is_working_day()` incorrectly identified certain Swedish holidays as working days.

According to Swedish Annual Leave Law (SFS 1977:480), Midsummer's Eve, Christmas Eve, and New Year's Eve must be treated equivalently to Sundays and public holidays for working day calculations, even though they are not official public holidays.

### Changes

- **New category**: Added `DE_FACTO` constant in `holidays/constants.py` for holidays that are legally treated like public holidays but are not officially designated as such
- **Sweden implementation**: 
  - Moved Midsummer's Eve (Midsommarafton), Christmas Eve (Julafton), and New Year's Eve (Nyårsafton) from `BANK` and `OPTIONAL` categories to the new `DE_FACTO` category
  - Added `_populate_de_facto_holidays()` method to handle these special holidays
  - Updated supported categories to include `DE_FACTO`
- **Documentation**: Added comprehensive documentation in `docs/holiday_categories.md` explaining the new category with usage examples
- **Localization**: Updated all Swedish locale files (en_US, sv, th, uk)
- **Tests**: Added comprehensive test coverage including specific `is_working_day()` validation
- **README**: Updated to reflect the new category availability for Sweden

### Breaking Change
**This is a MINOR breaking change for Sweden holidays with the OPTIONAL category.**
**Previous behavior:**
- Midsummer's Eve, Christmas Eve, and New Year's Eve were included in both BANK and OPTIONAL categories
**New behavior:**
- These three holidays have been moved to the new DE_FACTO category
- They are still included in BANK category (via _populate_de_facto_holidays() call)
- They are NO LONGER included in OPTIONAL category
**Impact:**
- ✅ Users relying on categories=BANK will **continue to see** these three holidays (no breaking change)
- ⚠️ Users relying on categories=OPTIONAL will **no longer see** these three holidays (breaking change)
- Users need to explicitly include DE_FACTO category for accurate working day calculations: Sweden(categories=(PUBLIC, DE_FACTO))

**Rationale:**
These holidays are not optional holidays—they are legally mandated to be treated like public holidays for working day calculations per Swedish Annual Leave Law (SFS 1977:480). The Riksbank (Swedish central bank) also recognizes them as bank holidays.

### Usage

For accurate working day calculations in Sweden, use:

```python
import holidays
from holidays.constants import PUBLIC, DE_FACTO

se = holidays.Sweden(categories=(PUBLIC, DE_FACTO), years=2024)
print(se.is_working_day('2024-12-24'))  # False (Christmas Eve)
print(se.is_working_day('2024-12-31'))  # False (New Year's Eve)
print(se.is_working_day('2024-06-21'))  # False (Midsummer's Eve)
```

### Legal Reference

Per Swedish Annual Leave Law (SFS 1977:480, Section 7):
> "Med söndag jämställs allmän helgdag samt midsommarafton, julafton och nyårsafton"
> 
> (Sundays are equivalent to public holidays as well as Midsummer's Eve, Christmas Eve, and New Year's Eve)

Source: https://www.riksdagen.se/sv/dokument-och-lagar/dokument/svensk-forfattningssamling/semesterlag-1977480_sfs-1977-480/

Fixes #3118

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Breaking change (a code change causing existing functionality to break)
- [x] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md).
- [x] I've run `make check` locally; all checks and tests passed.

vruss committed:
Add DE_FACTO holiday category for Sweden's legally-recognized non-public holidays

PPsyrius commented on holidays/countries/sweden.py:
```diff
@@ -126,6 +148,23 @@ def _populate_public_holidays(self):
+        day calculations according to Swedish Annual Leave Law (SFS 1977:480),
+        but are not official public holidays.
+        """
+        # Midsummer Eve (Friday between June 19 and 25).
```
```suggestion
        # Midsummer Eve.
```
This and rerun `make l10n` again to auto-address this (or just `make check`)

KJhellico commented on tests/countries/test_sweden.py:
```diff
@@ -173,6 +175,79 @@ def test_christmas(self):
     def test_sundays(self):
         self.assertSundays(Sweden)  # Sundays are considered holidays in Sweden.
 
+    def test_de_facto_holidays(self):
```
There is no need for separate methods for categories; full coverage in the methods for individual holidays is sufficient.

PPsyrius commented on tests/countries/test_sweden.py:
```diff
@@ -173,6 +175,79 @@ def test_christmas(self):
+        )
+
+        # New Year's Eve - every year
+        self.assertDeFactoHolidayName(name_newyear, (f"{year}-12-31" for year in self.full_range))
```
IMO this is currently a duplication of the existing `test_midsummer_eve`, `test_christmas_eve` and `test_new_years_eve` - we can safely remove this
```suggestion
```

PPsyrius commented on tests/countries/test_sweden.py:
```diff
@@ -173,6 +175,79 @@ def test_christmas(self):
+
+        # Public holidays should still be non-working
+        self.assertFalse(se_all.is_working_day("2024-12-25"))  # Juldagen
+        self.assertFalse(se_all.is_working_day("2024-01-01"))  # Nyårsdagen
```
Let's remove it in that case then

vruss commented on holidays/countries/sweden.py:
```diff
@@ -138,20 +177,11 @@ def _populate_common(self):
-        self._add_new_years_eve(tr("Nyårsafton"))
-
     def _populate_bank_holidays(self):
         self._populate_common()
```
Should also include defacto holidays inside optional?

vruss committed:
Add DE_FACTO holidays to BANK category per Riksbank guidance

- Include de facto holidays in bank holidays via _populate_de_facto_holidays()
- Update tests to reflect de facto holidays in BANK category
- Simplify comment in _populate_de_facto_holidays()

vruss committed:
Updated Project-Id-Version to 0.88 as per CR request

vruss commented:
Made sure to populate the bank days with defacto too. Optional ones are not being populated with DE_FACTO yet. If we also were to populate optional then I would question why DE_FACTO is even necessary in the first place since they would always be added.

vruss committed:
Updated test and added period at the end of Midsummer

PPsyrius reviewed (dismissed):
LGTM 🇸🇪 

Feel free to add your own name to [`CONTRIBUTORS`](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTORS) as well 🎉

vruss committed:
Added my name to CONTRIBUTORS

Signed-off-by: Viktor Rosvall <viktor.rosvall@monitor.se>

KJhellico reviewed (approved):
LGTM.

arkid15r reviewed (commented):
@vruss thanks for taking care of this 👍
